### DISCUSSION TITLE
partial, partialRight

### DISCUSSION THREAD

**Discussion by viveleroi** [remeda/remeda#957] [Suggest a Function]
We're hoping to replace lodash and ramda entirely with remeda. There are a few functions remeda doesn't have that I believe it could benefit from.

Lodash and ramda both have `partial`, and `partialRight` methods and I don't see any replacements in remeda yet. We use lodash's `parseInt` which is sort of a partialRight of `Number.parseInt` with the radix of 10 already applied. I was going to write my own but noticed there's no partial/currying methods yet.

[Lodash docs](https://lodash.com/docs/4.17.15#partial)
[Lodash source](https://github.com/lodash/lodash/blob/main/src/partial.ts)

**Comment by eranhirsch** [comment]
I'll be happy to look at any PR that does this. If you want to put less work upfront with the risk of getting rejects you can send a PR with just the logic and types (without the tests) so we can discuss over the PR.

**Comment by cjquines** [comment]
along these lines, we can also add:

- `curry` and `curryRight` like [lodash](https://lodash.com/docs/4.17.15#curry) or [ramda](https://ramdajs.com/docs/#curry). interestingly enough this can be strongly typed, see [this library](https://github.com/TomokiMiyauci/curry/blob/main/curry.ts)
- `partialObject` like [ramda](https://ramdajs.com/docs/#partialObject); implementation is easy but having the typings would be nice, maybe?

**Comment by eranhirsch** [comment]
@cjquines - this is actually done, right?
